backend/src/org/commcare/xml/CaseXmlParser.java

```diff
@@ -184,8 +186,8 @@ public Case parse() throws InvalidStructureException, IOException, XmlPullParser
 
                     String value = parser.nextText().trim();
 
-                    if(value.equals(caseId)) {
-                        throw new InvalidStructureException("Invalid index operation! Case " + caseId + " cannot index itself!", parser);
+                    if (value.equals(caseId)) {
+                        throw new ActionableInvalidStructureException("case.error.self.index", new String[]{caseId}, "Case " + caseId + " cannot index itself");
```

Comment:
The localization is performed at the javarosa level. The key only exists at the android level. I've updated the javarosa localization to return the default message if the localization key isn't found.

Where would we add the localization key to the J2ME code-base?